docs/proposals/ose-rebase.md

```diff
@@ -21,26 +24,175 @@ There are different types of commits that need to be carried on top of Origin ma
+commit message to:
+```
+[SQUASH][BRANDING] Branding changes in the oauth templates
+```
+The commit name is now valid and the PR can be merged thus resulting in the following set
+of carried commits in OSE master:
+```
+sge53da [SQUASH][BRANDING] Branding changes in the oauth templates
+fvda4fs [SQUASH][BRANDING] Updated branding code
+aaed106 Automatic commit of package [atomic-openshift] release [3.6.17-1].
+6a27890 [CARRY][BUILD_GEN] Specfile updates
+9bcbe1d [CARRY][BUILD] Tooling updates
+1abdc18 [CARRY][BRANDING] Branding updates
+```
+Note that you can have as many `SQUASH` commits with the same type as possible.
+
+Bob, again, needs to check in a new script specific to OSE so he creates a new commit with
+the script:
+```sh
+$ git add hack/set-foo.sh
+$ git commit -m "[SQUASH][SCRIPT] script for setting OSE-specific envs"
+```
+The new set of carries in Bob's branch is:
+```
+d4sf5fg [SQUASH][SCRIPT] script for setting OSE-specific envs
+sge53da [SQUASH][BRANDING] Branding changes in the oauth templates
+fvda4fs [SQUASH][BRANDING] Updated branding code
+aaed106 Automatic commit of package [atomic-openshift] release [3.6.17-1].
+6a27890 [CARRY][BUILD_GEN] Specfile updates
+9bcbe1d [CARRY][BUILD] Tooling updates
+1abdc18 [CARRY][BRANDING] Branding updates
+```
+Bob realizes just before he starts the test job for his PR that he got it wrong based on
+[rule #4](#naming-requirements). He steps back for a moment to think if his commit needs
+to squash into a pre-exisiting `[CARRY]` commit or deserves to be its own `[CARRY]`. After
+discussing it with Jane, they conclude that the new script can be squashed in the
+`[CARRY][BUILD]` commit. Bob updates the commit message and his PR gets merged. The state
+of master ends up being:
+```
+nphdbp9 [SQUASH][BUILD] script for setting OSE-specific envs
+sge53da [SQUASH][BRANDING] Branding changes in the oauth templates
+fvda4fs [SQUASH][BRANDING] Updated branding code
+aaed106 Automatic commit of package [atomic-openshift] release [3.6.17-1].
+6a27890 [CARRY][BUILD_GEN] Specfile updates
+9bcbe1d [CARRY][BUILD] Tooling updates
+1abdc18 [CARRY][BRANDING] Branding updates
+```
+
+Read below for the automated process that will collapse all the additional `[SQUASH]` commits
+created by Bob and Jane.
 
 ### The process
 
-During the actual rebase, `GIT_SEQUENCE_EDITOR` is going to be used by passing it a script that
-will automatically handle all the logic of dropping and squashing commits in long-term carries
-as necessary. If merge conflicts arise, they will need to be manually resolved as it already is
-the case today. In the future, we will extend the conflict resolution mechanism in Jenkins to
-send an e-mail to the author and committer of the conflicting commits.
+OSE builds run weekly every Monday, Wednesday, and Friday with the possibility to switch into
+daily builds. During an OSE build, the code is `git rebase`d on top of Origin master.
```

Comment:
I'm proposing to dropping git rebase, in favor of just rebase word to be consistent with what we say elsewhere that we do rebase. No need to get into how this is done at tech level.